jwql/website/apps/jwql/views.py

```diff
@@ -258,6 +262,100 @@ def home(request):
+
+    mnemonic_result = {}
+    mnemonic_query_result = {}
+    mnemonic_query_result_table_html = {}
+    mnemonic_query_result_plot = None
+
+    # If this is a POST request, we need to process the form data
+    if request.method == 'POST':
+
+        if 'mnemonic_name_search' in request.POST.keys():
+            mnemonic_name_search_form = MnemonicSearchForm(request.POST,
+                                                           prefix='mnemonic_name_search')
+
+            if mnemonic_name_search_form.is_valid():
+                mnemonic_identifier = mnemonic_name_search_form['search'].value()
+                if mnemonic_identifier is not None:
+                    mnemonic_result = query_mnemonic_info(mnemonic_identifier)
+
+            # create forms for search fields not clicked
+            form = FileSearchForm(prefix='file_search')
+            mnemonic_query_form = MnemonicQueryForm(prefix='mnemonic_query')
+
+        elif 'mnemonic_query' in request.POST.keys():
+            mnemonic_query_form = MnemonicQueryForm(request.POST, prefix='mnemonic_query')
+
+            # proceed only if entries make sense
+            if mnemonic_query_form.is_valid():
+                mnemonic_identifier = mnemonic_query_form['search'].value()
+                start_time = Time(mnemonic_query_form['start_time'].value(), format='iso')
+                end_time = Time(mnemonic_query_form['end_time'].value(), format='iso')
+
+                if mnemonic_identifier is not None:
+                    mnemonic_query_result = query_single_mnemonic(mnemonic_identifier, start_time,
+                                                                  end_time)
+                    import tempfile
+                    import sys
+                    tmpdir = tempfile.mkdtemp()
+                    mnemonic_query_result_table_html = os.path.join(tmpdir, 'mnemonic_query_result_table.html')
+                    print(mnemonic_query_result_table_html)
+                    # with open(mnemonic_query_result_table_html, 'w') as tmp:
+                    #     mnemonic_query_result.data.write(tmp, format='jsviewer')
+                    mnemonic_query_result.data[0:2].write(mnemonic_query_result_table_html, format='html')
+                    mnemonic_query_result.data[0:2].write(sys.stdout, format='html')
+                    # mnemonic_query_result.data.write(mnemonic_query_result_table_html, format='jsviewer')
+                    mnemonic_query_result_plot = mnemonic_query_result.bokeh_plot()
+                    print(mnemonic_query_result_table_html)
+
+            # create forms for search fields not clicked
+            form = FileSearchForm(prefix='file_search')
+            mnemonic_name_search_form = MnemonicSearchForm(prefix='mnemonic_name_search')
+
+        elif 'filesearch' in request.POST.keys():
+            # Create a form instance and populate it with data from the request
+            form = FileSearchForm(request.POST, prefix='file_search')
```

Comment:
Are you hoping to include a `FileSearchForm` on this webpage, or is this a relic from the code for the form on the homepage? (It seems like the `form` key in the `context` dictionary isn't getting used.)

Reply:
Removed. This was left over from development.